File: CodeChef/Self Learning/Indivisible_Permutation.java
//https://www.codechef.com/LP2TO301/problems/INDIPERM
import java.util.*;

class Indivisible_Permutation {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
     
        int t=sc.nextInt();
        while(t>0){
          
          int N=sc.nextInt();
          for(int i=1; i<N; i++){
             System.out.print((i+1) + " ");
          }
          System.out.println(1);
          t--;
       }
    }
    
}

File: CodeChef/Self Learning/Basic Programming Concepts/Indivisible_Permutation.cpp
//https://www.codechef.com/LP2TO301/problems/INDIPERM
#include<iostream>
using namespace std;

int main(){
    
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin>>t;

    while(t--)
    {
      int N;
      cin>>N;
      
      for(int i=1; i<N; i++){
         cout<<i+1<<" ";
      }
     cout<<1<<"\n";
    } 

    return 0;
}